src/replication.c

```diff
@@ -2476,17 +2485,22 @@ void replicationSetMaster(char *ip, int port) {
     int was_master = server.masterhost == NULL;
 
     sdsfree(server.masterhost);
-    server.masterhost = sdsnew(ip);
-    server.masterport = port;
+    server.masterhost = NULL;
     if (server.master) {
         freeClient(server.master);
     }
     disconnectAllBlockedClients(); /* Clients blocked in master, now slave. */
 
+    /* Setting masterhost only after the call to freeClient since it calls
```

Comment:
@oranagra So maybe it's better to set it to `NULL`, call `freeClient()` to clean up, then set the new masterhost and reconnect?

Reply:
@yossigo that's what i did.
or am i missing something?